### ISSUE TITLE
Unable to test response status codes with 3.0.0 version

### ISSUE THREAD

**Issue by jkryl** [chaijs/chai-http#104]
This fix renamed property in http response object from "statusCode" to "status": https://github.com/chaijs/chai-http/commit/1c5194f721d8bb1647f5f8162ff6dba7c5be4f9c.

It broke my tests which run with restify client and node version 4.4.0. And even the documentation for latest node version (6.x) says that the name of the property in response object should be "statusCode" (https://nodejs.org/api/http.html#http_response_statuscode). I don't understand to that fix. How can we just use a different property name for getting status code from response object and expect that everything will work as before?

**Comment by keithamus** [comment]
Hey @jkryl - thanks for the issue.

I think we got to here because SuperAgent - the underlying library we use, supports `status` ([see here](https://github.com/visionmedia/superagent/blob/b47a0112bc5d789e496f3f709a395125bf8537d7/lib/node/response.js#L173)). So it works fine for responses from SuperAgent.

We should have standard http responses still work, so I suggest we alter [the `status` assertion](https://github.com/chaijs/chai-http/blob/master/lib/http.js#L79-L90) so that it looks for _either_ `status` or `statusCode`. So something like....

```js
Assertion.addMethod('status', function (code) {
  var status = this._obj.status || this._obj.statusCode;

  this.assert(
      status == code
    , 'expected #{this} to have status code #{exp} but got #{act}'
    , 'expected #{this} to not have status code #{act}'
    , code
    , status
  );
});
```

If you'd like to make a PR with this change, I'd happily look through it and merge it, providing their are good tests to back it up. You'll also get [a place in our hall of fame](https://github.com/chaijs/chai-http/graphs/contributors) for being one of our amazing supercool contributors!

**Comment by meeber** [comment]
Resolved by #123
